sam/models/lasso_model.py

```diff
@@ -0,0 +1,226 @@
+        setting this to 'median' results in fitting that node with MAE (equal to 0.5 quantile).
+    feature_engineering: object, optional (default=None)
+        Should be an sklearn-type transformer that has a transform method, e.g.
+        `sam.feature_engineering.SimpleFeatureEngineer`.
+    alpha : float, default=1.0
+        Regularization constant that multiplies the L1 penalty term.
+    fit_intercept : bool, default=True
+        Whether or not to fit the intercept.
+    quantile_options : dict, optional (default=None)
+        Options for `sklearn.linear_model.QuantileRegressor`.
+    mean_options : dict, optional (default=None)
+        Options for `sklearn.linear_model.Lasso`.
+    kwargs: dict, optional
+        Not used. Just for compatibility of models that inherit from this class.
+
+    Attributes
+    ----------
+    feature_engineer_: Sklearn transformer
+        The transformer used on the raw data before prediction
+    n_inputs_: integer
+        The number of inputs used for the underlying neural network
+    n_outputs_: integer
+        The number of outputs (columns) from the model
+    prediction_cols_: array of strings
+        The names of the output columns from the model.
+    model_ : object
+        List of sklearn models, one for each quantile.
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> from sam.models import MLPTimeseriesRegressor
+    >>> from sam.feature_engineering import SimpleFeatureEngineer
+
+    >>> data = pd.read_parquet("../data/rainbow_beach.parquet").set_index("TIME")
+    >>> X, y = data, data["water_temperature"]
+
+    >>> simple_features = SimpleFeatureEngineer(keep_original=False)
+    >>> model = MLPTimeseriesRegressor(
+    ...     predict_ahead=(0,),
+    ...     feature_engineer=simple_features,
+    ... )
+    >>> model.fit(X, y)
+
+    """
+
+    def __init__(
+        self,
+        predict_ahead: Sequence[int] = (0,),
+        quantiles: Sequence[float] = (),
+        use_diff_of_y: bool = False,
+        timecol: str = None,
+        y_scaler: TransformerMixin = None,
+        average_type: str = "mean",
+        feature_engineer: BaseFeatureEngineer = None,
+        alpha: float = 1.0,
+        fit_intercept: bool = True,
+        quantile_options: dict = None,
+        mean_options: dict = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(
+            predict_ahead=predict_ahead,
+            quantiles=quantiles,
+            use_diff_of_y=use_diff_of_y,
+            timecol=timecol,
+            y_scaler=y_scaler,
+            feature_engineer=feature_engineer,
+            **kwargs,
+        )
+        self.average_type = average_type
+        self.alpha = alpha
+        self.fit_intercept = fit_intercept
+        self.quantile_options = quantile_options
+        self.mean_options = mean_options
+        self.feature_engineer = feature_engineer
+
+        if self.average_type == "median" and 0.5 in self.quantiles:
```

Comment:
Maybe we should add this check to the basemodel? We always do this check right?

Reply:
Not for `ConstantTimeseriesRegressor`. I could change it, or make an issue for it.

Reply:
Maybe make it an issue, so we don't forget